### PR TITLE
Group resources wip

### PR DESCRIPTION
**Pull request by oneswig**
Needs some thought on documentation and backwards compatibility...

### CODE REVIEW

File: files/environments/instance-library.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+resource_registry:
+  Cluster::InstanceWithVolume: ../resources/instance-w-volume.yaml
+  Cluster::Instance: ../resources/instance.yaml
+
+  # Under development
+  Cluster::InstanceMultiNet: ../resources/instance-multi.yaml
```
Comment (brtknr) [inline files/environments/instance-library.yaml]: I'm assuming this should be part of a separate PR

File: files/resources/instance-multi.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      #networks:
+      #  yaql:
+      #    data:
+      #      group_node_ports: { get_param: group_node_ports }
+      #      node_idx: { get_param: node_idx }
+      #    expression: $.data.group_node_ports[$.data.node_idx]
+      #
+      # FIXME: this works but is obviously limited to one port:
+      networks:
+        - port: { get_param: [ group_node_ports, {get_param: node_idx}, 0, port ] }
```
Comment (markgoddard) [inline files/resources/instance-multi.yaml]: Still needs fixing
